File: src/2.Modules/Billing/JadeCapital.Billing.PublicApi/Contracts/PlanInfo.cs
namespace JadeCapital.Billing.PublicApi.Contracts;

/// <summary>
/// Public DTO for the plan catalog. Wave-1.3 — fed to the marketing pricing
/// pages on the frontend (no auth required). Mirrors the wire shape consumed
/// by <c>frontend/src/app/core/api/plan-api.service.ts</c>.
///
/// Money is decimal (no float/double ever) and carries its currency code so
/// the frontend can render a localised price prefix without losing
/// precision.
/// </summary>
public sealed record PlanInfo(
    string Code,
    string Name,
    decimal MonthlyPrice,
    string Currency,
    bool IsEligibleForSelfService);

File: frontend/src/app/core/api/plan-info.ts
// ============================================================================
//  PlanInfo — DTO mirrored from
//  src/2.Modules/Billing/JadeCapital.Billing.PublicApi/Contracts/PlanInfo.cs
//
//  Wire shape from GET /api/billing/plans (Wave-1.3). AllowAnonymous
//  endpoint; the frontend consumes this BEFORE the user authenticates, so
//  the marketing pricing pages render the same prices the Admin catalog
//  controls (no more hardcoded PLANS arrays).
// ============================================================================

export interface PlanInfo {
  code: string;
  name: string;
  /** Monthly price in the plan's currency. Decimal on the backend, number here. */
  monthlyPrice: number;
  /** ISO-4217 3-letter currency code (e.g. "USD"). */
  currency: string;
  isEligibleForSelfService: boolean;
}
